src/cryptography/hazmat/primitives/asymmetric/rsa.py

```diff
@@ -42,13 +42,30 @@ def public_key(self):
 
 
 @six.add_metaclass(abc.ABCMeta)
-class RSAPrivateKeyWithNumbers(RSAPrivateKey):
+class RSAPrivateKeyWithSerialization(RSAPrivateKey):
```

Comment:
Is there really a reason that this needs to be a subclass?  Practically speaking, things which implement key serialization will also themselves be a key, but for testing or format-translation or other esoteric use-cases, mightn't there be a serialization provider with no actual `decrypt`?  Is there a reason to *prevent* that?